frontend/src/App.test.tsx

```diff
@@ -871,183 +996,150 @@ describe("App.handlePageInfoChanged", () => {
-      window.history,
-      // @ts-ignore
-      "pushState"
-    )
+    wrapper = shallow(<App {...getProps()} />)
+    instance = wrapper.instance() as App
+    // @ts-ignore
+    instance.sendBackMsg = jest.fn()
+    // @ts-ignore
+    instance.connectionManager.getBaseUriParts = mockGetBaseUriParts()
   })
 
   afterEach(() => {
     window.history.pushState({}, "", "/")
   })
 
-  it("figures out pageName when sendRerunBackMsg isn't given one (case 1: main page)", () => {
-    const props = getProps()
-    const wrapper = shallow(<App {...props} />)
-    const instance = wrapper.instance() as App
-    // @ts-ignore
-    instance.sendBackMsg = jest.fn()
-    // @ts-ignore
-    instance.connectionManager.getBaseUriParts = mockGetBaseUriParts()
-
-    instance.sendRerunBackMsg()
+  it("sends the pageScriptHash if one is given", () => {
+    instance.sendRerunBackMsg(undefined, "some_page_hash")
 
     // @ts-ignore
     expect(instance.sendBackMsg).toHaveBeenCalledWith({
-      rerunScript: { pageName: "", queryString: "" },
+      rerunScript: {
+        pageScriptHash: "some_page_hash",
+        pageName: "",
+        queryString: "",
+      },
     })
-
-    expect(wrapper.find("AppView").prop("currentPageName")).toEqual("")
   })
 
-  it("figures out pageName when sendRerunBackMsg isn't given one (case 2: non-main page)", () => {
-    const props = getProps()
-    const wrapper = shallow(<App {...props} />)
-    const instance = wrapper.instance() as App
-    // @ts-ignore
-    instance.sendBackMsg = jest.fn()
-    // @ts-ignore
-    instance.connectionManager.getBaseUriParts = mockGetBaseUriParts()
-
-    // Set the value of document.location.pathname to '/page1/' (with leading
-    // and trailing slashes to test that they get stripped)
-    window.history.pushState({}, "", "/page1/")
-
+  it("sends the currentPageScriptHash if no pageScriptHash is given", () => {
+    wrapper.setState({ currentPageScriptHash: "some_other_page_hash" })
     instance.sendRerunBackMsg()
 
     // @ts-ignore
     expect(instance.sendBackMsg).toHaveBeenCalledWith({
-      rerunScript: { pageName: "page1", queryString: "" },
+      rerunScript: {
+        pageScriptHash: "some_other_page_hash",
+        pageName: "",
+        queryString: "",
+      },
     })
-    expect(wrapper.find("AppView").prop("currentPageName")).toEqual("page1")
   })
 
-  it("figures out pageName when sendRerunBackMsg isn't given one and a baseUrlPath is set", () => {
-    const wrapper = shallow(<App {...getProps()} />)
-    const instance = wrapper.instance() as App
-    // @ts-ignore
-    instance.sendBackMsg = jest.fn()
-    // @ts-ignore
-    instance.connectionManager.getBaseUriParts = mockGetBaseUriParts("foo/bar")
-    instance.clearAppState = jest.fn()
-
-    // Set the value of document.location.pathname to '/foo/bar/page1'
-    window.history.pushState({}, "", "/foo/bar/page1")
-
+  it("extracts the pageName correctly if we can't get a pageScriptHash (main page)", () => {
```

Comment:
What is the correct behavior? It seems like an empty string is correct(?) 

I suggest renaming it to "extracts the pageName to an empty string if we can't get a pageScriptHash (main page)".